src/services/refactors/convertParamsToDestructuredObject.ts

```diff
@@ -272,15 +291,12 @@ namespace ts.refactor.convertParamsToDestructuredObject {
         if (!isValidParameterNodeArray(functionDeclaration.parameters, checker)) return false;
         switch (functionDeclaration.kind) {
             case SyntaxKind.FunctionDeclaration:
+                return hasNameOrDefault(functionDeclaration) && !!functionDeclaration.body && !checker.isImplementationOfOverload(functionDeclaration);
             case SyntaxKind.MethodDeclaration:
-                return !!functionDeclaration.name
-                    && !!functionDeclaration.body
-                    && !checker.isImplementationOfOverload(functionDeclaration);
+                return !!functionDeclaration.body && !checker.isImplementationOfOverload(functionDeclaration);

```

Comment:
>!!functionDeclaration.body && !checker.isImplementationOfOverload(functionDeclaration) [](start = 23, length = 86)

This check appears to be repeated in three of the cases.  Consider extracting a helper.